src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1016,10 +1018,16 @@ protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator
                 _fieldNames.Clear();
                 CachedProperties = null; // need to release this so can be collected
             }
+
+            for (var i = 0; i < _numberOfAllocatedStringsValues; i++)
+                _allocateStringValues[i].Reset();
```

Comment:
Resseting not only allows to return the memory, but also to clear other fields, e.g. materialized string value